packages/core/src/reducers/activities.ts

```diff
@@ -92,12 +99,9 @@ function upsertActivityWithSort(activities: WebChatActivity[], nextActivity: Web
       !(nextClientActivityID && clientActivityID === nextClientActivityID) && !(id && id === nextActivity.id)
   );
 
-  // Then, find the right (sorted) place to insert the new activity at, based on timestamp
-  // Since clockskew might happen, we will ignore timestamp on messages that are sending
-
+  // Then, find the right (sorted) place to insert the new activity at, based on sequence ID.
   const indexToInsert = nextActivities.findIndex(
-    ({ channelData: { state, 'webchat:sequence-id': sequenceId } = {} }) =>
-      (sequenceId || 0) > (nextSequenceId || 0) && state !== SENDING && state !== SEND_FAILED
+    ({ channelData: { 'webchat:sequence-id': sequenceId } = {} }) => (sequenceId || 0) > (nextSequenceId || 0)
```

Comment:
All activities are already patched with `webchat:sequence-id`. Thus, we can just purely use sequence ID to sort activities, without any concern whether certain activities had been sent or not.